src/web/components/Contributor.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { css } from 'emotion';
+import TwitterIcon from '@frontend/static/icons/twitter.svg';
+import { palette } from '@guardian/src-foundations';
+import { headline, textSans } from '@guardian/src-foundations/typography';
+import { pillarPalette } from '@root/src/lib/pillars';
+
+const twitterHandle = css`
+    ${textSans.xsmall()};
+    font-weight: bold;
+    color: ${palette.neutral[46]};
+
+    padding-right: 10px;
+    display: inline-block;
+
+    svg {
+        height: 10px;
+        max-width: 12px;
+        margin-right: 0px;
+        fill: ${palette.neutral[46]};
+    }
+
+    a {
+        color: ${palette.neutral[46]};
+        text-decoration: none;
+    }
+`;
+
+const bylineStyle = (pillar: Pillar) => css`
+    ${headline.xxxsmall()};
+    color: ${pillarPalette[pillar].main};
+    padding-bottom: 8px;
+    font-style: italic;
+
+    a {
+        font-weight: 700;
+        color: ${pillarPalette[pillar].main};
+        text-decoration: none;
+        font-style: normal;
+        :hover {
+            text-decoration: underline;
+        }
+    }
+`;
+
+// This crazy function aims to split bylines such as
+// 'Harry Potter in Hogwarts' to ['Harry Potter', 'in Hogwarts']
+// Or
+// 'Jane Doe and John Smith` to ['Jane Doe', ' and ', 'John Smith']
+// It does this so we can have separate links to both contributors
+const bylineAsTokens = (bylineText: string, tags: TagType[]): string[] => {
+    const contributorTags = tags
+        .filter(t => t.type === 'Contributor')
+        .map(c => c.title);
+    // The contributor tag title should exist inside the bylineText for this regex to work
+    const regex = new RegExp(`(${contributorTags.join('|')})`);
+
+    return bylineText.split(regex);
+};
+
+const RenderContributor: React.FC<{
+    bylineText: string;
+    contributorTags: TagType[];
+    pillar: Pillar;
+}> = ({ bylineText, contributorTags, pillar }) => {
+    const renderedTokens = bylineAsTokens(bylineText, contributorTags).map(
+        (token, i) => {
+            const associatedTags = contributorTags.filter(
+                t => t.title === token,
+            );
+            if (associatedTags.length > 0) {
+                return (
+                    <ContributorLink
+                        contributor={token}
+                        contributorTagId={associatedTags[0].id}
+                        key={i}
+                    />
+                );
+            }
+            return token;
+        },
+    );
+
+    return <div className={bylineStyle(pillar)}>{renderedTokens}</div>;
+};
+
+const ContributorLink: React.FC<{
+    contributor: string;
+    contributorTagId: string;
+}> = ({ contributor, contributorTagId }) => (
+    <a
+        rel="author"
+        data-link-name="auto tag link"
+        href={`//www.theguardian.com/${contributorTagId}`}
```

Comment:
Do we not have a 'stage root' url from config for cases like this?

Reply:
Not sure, this code was just copied over. I checked the codebase and we do seem to be using 'www.theguardian.com' in a fair few places though.

I guess if we're confident this string won't change then hardcoding it is okay?

Reply:
More that it's useful if look at code or local, clicking links take you to the same place? Tech debt.

Reply:
Good point. If we introduced navigation we'd need this.

I'll add a card.

Reply:
https://trello.com/c/AyaaatOO/968-hardcoded-site-urls-as-variables